graphql-code-generator/src/main/java/io/ballerina/graphql/generator/service/combiner/ServiceTypesFileCombiner.java

```diff
@@ -0,0 +1,177 @@
+import io.ballerina.compiler.syntax.tree.IntersectionTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.ObjectTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.RecordTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.compiler.syntax.tree.UnionTypeDescriptorNode;
+import io.ballerina.graphql.generator.CodeGeneratorConstants;
+import io.ballerina.graphql.generator.service.comparator.ClassDefinitionComparator;
+import io.ballerina.graphql.generator.service.comparator.EnumDeclarationComparator;
+import io.ballerina.graphql.generator.service.comparator.TypeDefinitionComparator;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocuments;
+import org.ballerinalang.formatter.core.Formatter;
+import org.ballerinalang.formatter.core.FormatterException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createNodeList;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createToken;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createModulePartNode;
+
+/**
+ * Utility class for combining available service with generated service for a GraphQL schema.
+ */
+public class ServiceTypesFileCombiner {
+    private final ModulePartNode nextContentNode;
+    private final ModulePartNode prevContentNode;
+    private GraphQLSchema nextGraphqlSchema;
+    private List<ModuleMemberDeclarationNode> moduleMembers;
+    private List<ModuleMemberDeclarationNode> inputObjectTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> interfaceTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> enumTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> unionTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> objectTypesModuleMembers;
+    private List<String> breakingChangeWarnings;
+
+    public ServiceTypesFileCombiner(ModulePartNode prevContentNode, ModulePartNode nextContentNode,
+                                    GraphQLSchema nextGraphqlSchema) {
+        this.prevContentNode = prevContentNode;
+        this.nextContentNode = nextContentNode;
+        this.nextGraphqlSchema = nextGraphqlSchema;
+
+        moduleMembers = new ArrayList<>();
+        inputObjectTypesModuleMembers = new ArrayList<>();
+        interfaceTypesModuleMembers = new ArrayList<>();
+        enumTypesModuleMembers = new ArrayList<>();
+        unionTypesModuleMembers = new ArrayList<>();
+        objectTypesModuleMembers = new ArrayList<>();
+
+        breakingChangeWarnings = new ArrayList<>();
```

Comment:
When referring to class variables, use the `this` keyword. It will make the code clearer. Might not seem necessary in the constructor, but not using `this` will be messy inside other methods. So, as a practice, better to use it in all the places.